### ISSUE TITLE
Dimensions greater than 3

### ISSUE THREAD

**Issue by patrickocal** [gpufit/Gpuspline#5]
Hi, 

do you have any examples of NL regressions of dimension greater than 3? For example a 20-dimensional version of your "f" function in your Levenberg-Marquardt algorithm in the paper in Nature.

Thanks,
Patrick

**Comment by jkfindeisen** [comment]
No, unfortunately not. It could probably be done since the 2D and 3D spline methods rely on 1D splines in different directions, but it's currently not implemented and as far as I'm aware the code here is all there is. It would mean a bit more bookkeeping and the number of spline coefficients needed per interval would increase exponentially with the dimensionality.

In the publication, there are four channels of a 3D function each, which is still on the level of 3D splines.

**Comment by patrickocal** [comment]
Thanks for your answer! (Only just saw it now.)
